scipy/optimize/tests/test__dual_annealing.py

```diff
@@ -178,10 +178,14 @@ def test_local_search_option_bounds(self):
         func = lambda x: np.sum((x-5) * (x-1))
         bounds = list(zip([-6, -5], [6, 5]))
         # Test bounds can be passed (see gh-10831)
-        dual_annealing(
-            func,
-            bounds=bounds,
-            local_search_options={"method": "SLSQP", "bounds": bounds})
+
+        with np.testing.suppress_warnings() as sup:
+            sup.record(RuntimeWarning, "Values in x were outside bounds ")
```

Comment:
The reason this occurs is because SLSQP sometimes suggests trial solutions that are not with the bounds, i.e.:
`lb <= x <= ub` is not obeyed. It's only outside by a small amount, 1 or 2 ULP. This is an issue with the SLSQP Fortran code. The solution we came up with is to clip to the bounds; values on the bounds are acceptable.